### DISCUSSION TITLE
Example: FedAvg, Homomorphic Encryption with flat file data

### DISCUSSION THREAD

**Discussion by manzurelahi** [NVIDIA/NVFlare#1990] [General]
Hi
I am learning NVFlare and am new to this platform.
I aim to run Federated Learning: FedAvg algorithm, Homomorphic Encryption with Health Data in Flat file /tabular format.

I am currently using Azure Ubuntu Virtual Machine with 2 GPU.
Do you have any suggestions, guidelines, or examples I can look at?

So far, I have successfully run the [Hello PyTorch](https://github.com/NVIDIA/NVFlare/blob/main/examples/hello-world/hello-pt/README.md) with CIFAR-10. This is an excellent example of FedAvg and the image data model.

The closest example I can find is the [Real-World Federated Learning](https://github.com/NVIDIA/NVFlare/blob/main/examples/advanced/cifar10/cifar10-real-world/README.md) with CIFAR-10, which has a Homomorphic Encryption example. But I am having some issues with it. I do not need real-world set up at this stage
I tried to modify the [Simulated Federated Learning](https://github.com/NVIDIA/NVFlare/blob/main/examples/advanced/cifar10/cifar10-sim/README.md) with CIFAR-10 as per Real-World Federated Learning to include Homomorphic Encryption. (changed config_fed_client and config_fed_server file but no luck)

Any assistance would be highly appreciated.
Thanks in advance.

**Comment by holgerroth** [comment]
Seems your task is similar to the one in this [discussion](https://github.com/NVIDIA/NVFlare/discussions/1994). Please see there for some hints. Currently, we don't support running HE out of the box with the simulator as the encryption keys need to be provisioned first as shown in the real-world example.
